pkg/producer/kafka.go

```diff
@@ -0,0 +1,182 @@
+				flowMsg.ReversePacketsTotal = ie.Value.(uint64)
+			case "reverseOctetTotalCount":
+				flowMsg.ReverseBytesTotal = ie.Value.(uint64)
+			case "reversePacketDeltaCount":
+				flowMsg.ReversePacketsDelta = ie.Value.(uint64)
+			case "reverseOctetDeltaCount":
+				flowMsg.ReverseBytesDelta = ie.Value.(uint64)
+			case "sourcePodNamespace":
+				flowMsg.SrcPodNamespace = ie.Value.(string)
+			case "sourcePodName":
+				flowMsg.SrcPodName = ie.Value.(string)
+			case "sourceNodeName":
+				flowMsg.SrcNodeName = ie.Value.(string)
+			case "destinationPodNamespace":
+				flowMsg.DstPodNamespace = ie.Value.(string)
+			case "destinationPodName":
+				flowMsg.DstPodName = ie.Value.(string)
+			case "destinationNodeName":
+				flowMsg.DstNodeName = ie.Value.(string)
+			case "destinationClusterIPv4", "destinationClusterIPv6":
+				if flowMsg.DstClusterIP != "" {
+					klog.Warningf("Do not expect destination IP: %v to be filled already", flowMsg.DstClusterIP)
+				}
+				flowMsg.DstClusterIP = ie.Value.(net.IP).String()
+			case "destinationServicePort":
+				flowMsg.DstServicePort = uint32(ie.Value.(uint16))
+			case "destinationServicePortName":
+				flowMsg.DstServicePortName = ie.Value.(string)
+			case "ingressNetworkPolicyName":
+				flowMsg.IngressPolicyName = ie.Value.(string)
+			case "ingressNetworkPolicyNamespace":
+				flowMsg.IngressPolicyNamespace = ie.Value.(string)
+			case "egressNetworkPolicyName":
+				flowMsg.EgressPolicyName = ie.Value.(string)
+			case "egressNetworkPolicyNamespace":
+				flowMsg.EgressPolicyNamespace = ie.Value.(string)
+			default:
+				klog.Warningf("There is no field with name: %v in flow message (.proto schema)", ie.Element.Name)
+			}
+		}
+		flowMsgs = append(flowMsgs, flowMsg)
+	}
+	return flowMsgs
+}
+
+func NewKafkaProducer(asyncProducer sarama.AsyncProducer, topic string) *KafkaProducer{
+	return &KafkaProducer{
+		producer: asyncProducer,
+		topic:    topic,
+	}
+}
+
+// InitKafkaProducer with broker addresses and other Kafka config parameters.
+func InitKafkaProducer(addrs []string, topic string, logErrors bool) (*KafkaProducer, error) {
+	kafkaConfig := sarama.NewConfig()
+	kafkaConfig.Version = KafkaConfigVersion
+	kafkaConfig.Producer.Return.Successes = false
+	kafkaConfig.Producer.Return.Errors = logErrors
+
+	asyncProducer, err := sarama.NewAsyncProducer(addrs, kafkaConfig)
+	if err != nil {
+		return nil, err
+	}
+	producer := NewKafkaProducer(asyncProducer, topic)
+
+	// Capturing errors from Kafka sarama client
+	if logErrors {
+		go func() {
+			for msg := range asyncProducer.Errors() {
+				klog.Error(msg)
+			}
+		}()
+	}
+
+	return producer, nil
+}
+
+// SendFlowMessage takes in the flow message in proto schema, encodes it and sends
+// it to on the producer channel.
+func (kp *KafkaProducer) SendFlowMessage(msg *protobuf.FlowMessage) {
+	buf := proto.NewBuffer([]byte{})
+	if err := buf.EncodeMessage(msg); err != nil {
+		klog.Errorf("Error when encoding flow message: %v", err)
```

Comment:
Should we return here whe error happened?

Reply:
Yes, need to return. Thanks for catching. Done.